internal/controllers/certmanager.go

```diff
@@ -103,22 +104,42 @@ func (r *CryostatReconciler) setupTLS(ctx context.Context, cr *operatorv1beta1.C
 		return nil, err
 	}
 
-	// Create a certificate for Grafana signed by the Cryostat CA
-	grafanaCert := resources.NewGrafanaCert(cr)
-	err = r.createOrUpdateCertificate(ctx, grafanaCert, cr)
-	if err != nil {
-		return nil, err
-	}
-
 	// Create a certificate for the reports generator signed by the Cryostat CA
 	reportsCert := resources.NewReportsCert(cr)
 	err = r.createOrUpdateCertificate(ctx, reportsCert, cr)
 	if err != nil {
 		return nil, err
 	}
+	tlsConfig := &resources.TLSConfig{
+		CryostatSecret:     cryostatCert.Spec.SecretName,
+		ReportsSecret:      reportsCert.Spec.SecretName,
+		KeystorePassSecret: cryostatCert.Spec.Keystores.PKCS12.PasswordSecretRef.Name,
+	}
+	certificates := []*certv1.Certificate{caCert, cryostatCert, reportsCert}
+	// Create a certificate for Grafana signed by the Cryostat CA
+	if !cr.Spec.Minimal {
+		grafanaCert := resources.NewGrafanaCert(cr)
+		err = r.createOrUpdateCertificate(ctx, grafanaCert, cr)
+		if err != nil {
+			return nil, err
+		}
+		certificates = append(certificates, grafanaCert)
+		tlsConfig.GrafanaSecret = grafanaCert.Spec.SecretName
+	} else if cr.Spec.Minimal {
+		grafanaCert := resources.NewGrafanaCert(cr)
+		secret, err := r.GetCertificateSecret(ctx, grafanaCert)
```

Comment:
A couple of things here:
1. We should check `err` and return if not nil.
2. It seems like using `GetCertficateSecret` causes a problem when the secret does not exist, for example when it's already been deleted. We don't actually need the contents of the actual secret, we just need the name and namespace to delete it.
    Could I suggest adding something like the following function:
    ```go
    func secretForCertificate(cert *certv1.Certificate) *corev1.Secret {
    	return &corev1.Secret{
    		ObjectMeta: metav1.ObjectMeta{
    			Name:      cert.Spec.SecretName,
    			Namespace: cert.Namespace,
    		},
    	}
    }
    ```
    This would suffice instead of using `GetCertificateSecret`.